src/Essentials/src/FileSystem/FileSystem.uwp.cs

```diff
@@ -9,11 +9,21 @@ namespace Microsoft.Maui.Storage
 {
 	partial class FileSystemImplementation : IFileSystem
 	{
+		static string CleanPath(string path) =>
+			string.Join("_", path.Split(Path.GetInvalidFileNameChars()));
+
+		static string AppSpecificPath =>
+			Path.Combine(CleanPath(AppInfoImplementation.PublisherName), CleanPath(AppInfo.PackageName));
```

Comment:
In all honesty, this is not so great because the app ID is a GUID, which means the AppData folder will end up with a bunch of GUIDs. But, that is also what is in the `C:\ProgramData\Packages` folder and that is where all the MSIX apps put things. So not a machine issue - more a user issue.